src/main/java/org/assertj/core/api/Assumptions.java

```diff
@@ -1277,18 +1278,19 @@ public static OptionalLongAssert assumeThat(OptionalLong actual) {
   @SuppressWarnings("unchecked")
   private static <ASSERTION> Class<? extends ASSERTION> createAssumptionClass(Class<ASSERTION> assertClass) {
     SimpleKey cacheKey = new SimpleKey(assertClass);
-    return (Class<ASSERTION>) CACHE.findOrInsert(Assumptions.class.getClassLoader(),
+    return (Class<ASSERTION>) CACHE.findOrInsert(assertClass.getClassLoader(),
                                                  cacheKey,
                                                  () -> generateAssumptionClass(assertClass));
   }
 
   protected static <ASSERTION> Class<? extends ASSERTION> generateAssumptionClass(Class<ASSERTION> assertionType) {
+    ClassLoadingStrategyPair strategy = classLoadingStrategy(assertionType);
     return BYTE_BUDDY.subclass(assertionType)
                      // TODO ignore non assertion methods ?
                      .method(any())
                      .intercept(ASSUMPTION)
                      .make()
-                     .load(Assumptions.class.getClassLoader(), classLoadingStrategy(assertionType))
+                     .load(strategy.getClassLoader(), strategy.getClassLoadingStrategy())
```

Comment:
@bjhargrave can you add a test for these changes?

Reply:
Done given the limitations that one cannot assume on a custom assertion class.